### ISSUE TITLE
Same URL used in multiple SSL check

### ISSUE THREAD

**Issue by DB-CL** [Icinga/icinga2#8847]
## Describe the bug

It looks like I have a bug with the `apply Service` and a for loop

My service is defined like that : 
```
apply Service "SSL - " for (name => file in host.vars.urls_to_check) {
    import "generic-service"
    vars.ssl_address = file.url
    vars.ssl_cert_valid_days_warn = 5
    vars.ssl_cert_valid_days_critical = 1
    check_command = "ssl"

    assign where file.ssl
}
```

And I have a "urls_to_check" variable in my host file defined like that : 

```
vars.urls_to_check["sub1.example1.com"] = {
        url = "sub1.example1.com"
        ssl = true
            }
vars.urls_to_check["sub2.example2.com"] = {
        url = "sub2.example2.com"
        ssl = true
            }
```

But when it comes to checking, Icinga runs all the checks over "sub1.example1.com", even the service named "SSL - sub2.examle2.com".

The strangest part is that in the web interface, when I explore my "example2" service, I can see a variable named "Ssl address"

![image](https://user-images.githubusercontent.com/191572/123134912-90b2ab80-d451-11eb-9239-ac08c0598f9b.png)

And its value is "sub2.example2.com", but in the plugin output it's clearly using "sub1.example1.com"

## Expected behavior

The check of the sub2.example2.com should be run over sub2.example2.com and not sub1.example1.com

## Your Environment

Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`): r2.12.2-1
* Operating System and version: Debian Jessie
* Enabled features (`icinga2 feature list`): Enabled features: api checker command ido-pgsql influxdb mainlog notification
* Disabled features: compatlog debuglog elasticsearch gelf graphite icingadb ido-mysql livestatus opentsdb perfdata statusdata syslog
* Icinga Web 2 version and modules (System - About): 2.7.4
* Config validation (`icinga2 daemon -C`): Nothing relevant
* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes. 

## Additional context

I discovered it today but it's happening on all my hosts very constistently

**Comment by Al2Klimov** [comment]
Hello @DB-CL and thank you for reporting!

What means 'Icinga runs all the checks **over** "sub1.example1.com"'?

Best,
AK

**Comment by DB-CL** [comment]
Hi, 
Sorry for my bad english.
It means that even for the service "SSL - sub2.example2.com", it sends the http request on sub1.example1.com, never sub2.example2.com.

**Comment by Al2Klimov** [comment]
Which logs make you thinking so?

**Comment by DB-CL** [comment]
Hi, 
Sorry, I answered too quickly to last question. I thought I remembered the bug but I didnt.

I went back to check the logs and here is a better bug description :

If I have two subdomains on the same host, the SSL check is done on the wrong subdomain, and I can tell this by looking at the output of the plugin in my Icingaweb2 interface : 

Output for the service "SSL - sub1.example1.com "
```
OK - Certificate 'sub1.example1.com' will expire on Wed 25 Aug 2021 02:02:44 PM GMT +0000.
TCP OK - 0.022 second response time on sub1.example1.com port 443
```

Output for the service "SSL - sub2.example2.com"
```
OK - Certificate 'sub1.example1.com' will expire on Wed 25 Aug 2021 02:02:44 PM GMT +0000.
TCP OK - 0.022 second response time on sub2.example2.com port 443
```
(there is no mistake here : it's sub1.example1.com in the first line and sub2.example2.com in the second line)

And I discovered this bug because my sub2.example2.com expired so it's not a display error, the check isn't done on the right certificate.

When I access https://sub1.example1.com and https://sub2.example2.com with a browser, I have two different certificates.
